packages/rich-text/src/internal/types/editor.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * https://plate.udecode.io/docs/typescript
+ */
+import { MARKS } from '@contentful/rich-text-types';
+import * as p from '@udecode/plate-core';
+import * as s from 'slate';
+import * as sr from 'slate-react';
+import {
+  SelectionMoveOptions as SlateSelectionMoveOptions,
+  SelectionCollapseOptions as SlateSelectionCollapseOptions,
+} from 'slate/dist/transforms/selection';
+import { TextInsertTextOptions as SlateTextInsertTextOptions } from 'slate/dist/transforms/text';
+
+import { TrackingPluginActions } from '../../plugins/Tracking';
+
+export interface Text extends p.TText {
+  [MARKS.BOLD]?: boolean;
+  [MARKS.CODE]?: boolean;
+  [MARKS.ITALIC]?: boolean;
+  [MARKS.UNDERLINE]?: boolean;
+  [MARKS.SUPERSCRIPT]?: boolean;
+  [MARKS.SUBSCRIPT]?: boolean;
+}
+
+export interface Element extends p.TElement {
+  type: string;
+  data?: Record<string, unknown>;
+  isVoid?: boolean;
+  children: (Text | Element)[];
+}
+
+export type Value = Element[];
+export type ReactEditor = p.TReactEditor<Value>;
+export interface PlateEditor extends p.PlateEditor<Value> {
+  tracking: TrackingPluginActions;
+}
+
+export type CustomText = {
+  text: string;
+  [MARKS.BOLD]?: boolean;
+  [MARKS.CODE]?: boolean;
+  [MARKS.ITALIC]?: boolean;
+  [MARKS.UNDERLINE]?: boolean;
+  [MARKS.SUPERSCRIPT]?: boolean;
+  [MARKS.SUBSCRIPT]?: boolean;
+};
+
+export type TextOrCustomElement = CustomElement | CustomText;
+
+export type CustomElement<T = unknown> = {
+  type: string;
+  children: TextOrCustomElement[];
+  data: T;
+  isVoid?: boolean;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- TODO: explain this disable
+export type CustomRenderElementProps<T = any, O = any> = Omit<RenderElementProps, 'element'> & {
+  element: CustomElement<T>;
+} & O;
+
```

Comment:
❗  These are legacy types that should have been removed by the removal of `src/types.ts` file. `CustomText` is now `Text`, `CustomElement` is `Element` , `TextOrCustomELement` is simply the `Node` I think. And we should be using the `RenderElementsProps` type defined below.